### PR TITLE
ci: introduce markdown lint

### PR DESCRIPTION
**Pull request by An-DJ**
## What is the purpose of the change

Introduce markdown lint to check the doc format.

Reference: https://github.com/DavidAnson/markdownlint/tree/v0.32.1/doc

* `MD001/heading-increment` : Heading levels should only increment by one level at a time
* `MD003/heading-style` : Heading style
* `MD004/ul-style` : Unordered list style
* `MD005/list-indent` : Inconsistent indentation for list items at the same level
* `MD007/ul-indent` : Unordered list indentation
* `MD009/no-trailing-spaces` : No trailing spaces
* `MD010/no-hard-tabs` : Hard tab
* `MD011/no-reversed-links` : Reversed link syntax
* `MD012/no-multiple-blanks` : Multiple consecutive blank lines
* `MD022/blanks-around-headings` : Headings should be surrounded by blank lines
* `MD023/heading-start-left` : Headings must start at the beginning of the line
* `MD030/list-marker-space` : Spaces after list markers
* `MD031/blanks-around-fences` : Fenced code blocks should be surrounded by blank lines
* `MD032/blanks-around-lists` : Lists should be surrounded by blank lines
* `MD034/no-bare-urls` : Bare URL used
* `MD042/no-empty-links` : No empty links
* `MD046/code-block-style` : Code block style
* `MD047/single-trailing-newline` : Files should end with a single newline character
* `MD048/code-fence-style` : Code fence style
* `MD049/emphasis-style` : Emphasis style
* `MD050/strong-style` : Strong style

## Brief changelog

Introduce markdown lint to check the doc format.

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/seata/seata.github.io/issues) filed for the change(usually before you start working on it).  Trivial changes such as typos do not necessitate a GITHUB issue. Your pull request should solely focus on addressing this specific issue without incorporating any other modifications. Each PR should resolve a single issue.
- [ ] Format the pull request title like `doc: add an operations guide document`. Each commit within the pull request should contain a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Test your code locally by running `npm run build`, and make sure it works as expected.
- [ ] Ensure that no files under the `build` folder are included in the commit, as well as the `package-lock.json` and `yarn.lock` files.

### CODE REVIEW

**Review by slievrly — COMMENTED** [review]
CI failed.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  markdownlint:
+    name: Markdown lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20.10.0'
```
Comment (liuqiufeng) [inline .github/workflows/lint.yml:14]: nodejs版本统一使用19.5.x
The nodejs version is standardized to 19.5.x.

**Comment by liuqiufeng** [comment]
另外，需要考虑开发者如何在本地执行markdown lint修复

**Comment by An-DJ** [comment]
@liuqiufeng Currently, lint rules only work on `README.md` as a demo in this PR. We can easily get the lint check result:

```
$ markdownlint '**/*.md' --ignore src \
             --ignore node_modules \
             --ignore blog \
             --ignore i18n \
             --ignore versioned_docs \
             --ignore versioned_sidebars \
             --ignore docs \
             --ignore readme
```

Output:

```
README.md:5:25 MD034/no-bare-urls Bare URL used [Context: "https://seata.io"]
README.md:7 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "# README.md"]
README.md:7 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# README.md"]
README.md:8 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- en [English](README.md)"]
README.md:11 MD001/heading-increment Heading levels should only increment by one level at a time [Expected: h2; Actual: h3]
README.md:19 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "3. Visit"]
README.md:20 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:20 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:30 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "3. Visit"]
README.md:31 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:31 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:35 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## Requirements"]
README.md:39 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## Note for i18n"]
README.md:42 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## Overview of the Website Structure"]
README.md:46 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:46 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:88 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Directories Related to Adding Documents"]
README.md:105 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Summary of Adding Documents"]
README.md:120:1 MD027/no-multiple-space-blockquote Multiple spaces after blockquote symbol [Context: ">  Notes: Corresponding to Chi..."]
README.md:122 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
README.md:123 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## How To Add Blogs"]
README.md:126 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Directories Related to Adding Blogs"]
README.md:134 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Summary of Adding Blogs"]
README.md:141:1 MD027/no-multiple-space-blockquote Multiple spaces after blockquote symbol [Context: ">  Notes: Corresponding to Chi..."]
README.md:146 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:146 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
README.md:155 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
README.md:157:31 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
README.md:157 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "1. `title` can not include `:`"]
README.md:161 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## Note for .md file"]
README.md:162 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "1. Do not use un correct html ..."]
README.md:167:1 MD004/ul-style Unordered list style [Expected: dash; Actual: plus]
```

Obviously we just follow the output info and fix them one by one.

However, there are too many existed docs without beening checked by lint in other directories.

Then IMHO, we can ingore them temporarily in this PR, and fix them in new several PRs.

**Comment by liuqiufeng** [comment]
> @liuqiufeng Currently, lint rules only work on `README.md` as a demo in this PR. We can easily get the lint check result:
> 
> ```
> $ markdownlint '**/*.md' --ignore src \
>              --ignore node_modules \
>              --ignore blog \
>              --ignore i18n \
>              --ignore versioned_docs \
>              --ignore versioned_sidebars \
>              --ignore docs \
>              --ignore readme
> ```
> 
> Output:
> 
> ```
> README.md:5:25 MD034/no-bare-urls Bare URL used [Context: "https://seata.io"]
> README.md:7 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "# README.md"]
> README.md:7 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# README.md"]
> README.md:8 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "- en [English](README.md)"]
> README.md:11 MD001/heading-increment Heading levels should only increment by one level at a time [Expected: h2; Actual: h3]
> README.md:19 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "3. Visit"]
> README.md:20 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
> README.md:20 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
> README.md:30 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "3. Visit"]
> README.md:31 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
> README.md:31 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
> README.md:35 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## Requirements"]
> README.md:39 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## Note for i18n"]
> README.md:42 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## Overview of the Website Structure"]
> README.md:46 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
> README.md:46 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
> README.md:88 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Directories Related to Adding Documents"]
> README.md:105 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Summary of Adding Documents"]
> README.md:120:1 MD027/no-multiple-space-blockquote Multiple spaces after blockquote symbol [Context: ">  Notes: Corresponding to Chi..."]
> README.md:122 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
> README.md:123 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## How To Add Blogs"]
> README.md:126 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Directories Related to Adding Blogs"]
> README.md:134 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "### Summary of Adding Blogs"]
> README.md:141:1 MD027/no-multiple-space-blockquote Multiple spaces after blockquote symbol [Context: ">  Notes: Corresponding to Chi..."]
> README.md:146 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
> README.md:146 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
> README.md:155 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
> README.md:157:31 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
> README.md:157 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "1. `title` can not include `:`"]
> README.md:161 MD022/blanks-around-headings Headings should be surrounded by blank lines [Expected: 1; Actual: 0; Below] [Context: "## Note for .md file"]
> README.md:162 MD032/blanks-around-lists Lists should be surrounded by blank lines [Context: "1. Do not use un correct html ..."]
> README.md:167:1 MD004/ul-style Unordered list style [Expected: dash; Actual: plus]
> ```
> 
> Obviously we just follow the output info and fix them one by one.
> 
> However, there are too many existed docs without beening checked by lint in other directories.
> 
> Then IMHO, we can ingore them temporarily in this PR, and fix them in new several PRs.

目前的写法需要提交pr之后运行ci才能知道哪里写的有问题，或使用vscode的markdownlint插件，或者本地全局安装markdownlint。不太方便。
The current writeup needs to be submitted to pr and then run ci to see where the writeup is wrong,or using the markdownlint plugin for vscode,or local global installation of markdownlint package.not very convenient.
我的意思是，增加一个开发依赖markdownlint，再加一个npm命令，只需要跑一下就可以自动扫描或自动修复。
I mean, add a development dependency markdownlint and an npm command to just run it to autoscan or autofix.

**Comment by An-DJ** [comment]
> I mean, add a development dependency markdownlint and an npm command to just run it to autoscan or autofix.

@liuqiufeng Done. Now we can use `npm run lint` to conveniently check the doc style : )

**Review by liuqiufeng — APPROVED** [review]
LGTM

**Comment by An-DJ** [comment]
Need workflow approval @liuqiufeng @slievrly broken ci fixed.

**Review by funky-eyes — COMMENTED** [review]
将seata.io都改为seata.apache.org
Replace all seata.io with seata.apache.org

**Comment by An-DJ** [comment]
> 将seata.io都改为seata.apache.org Replace all seata.io with seata.apache.org

@funky-eyes done.

**Review by funky-eyes — APPROVED** [review]
LGTM

**Pull request merged**